### ISSUE TITLE
AttributeError: module 'pulp' has no attribute 'HiGHS_CMD'

### ISSUE THREAD

**Issue by ssurbhi560** [ERGO-Code/HiGHS#897]
Apologies if this might not be the right place to ask this. But I have been trying to run a Jupyter Notebook application which used  HiGHS,

```
highs_path = "/Users/surbhisharma/Downloads/voila-authentication/WDN_Scheduler/HiGHS/build/bin/highs"
solver_highs = pl.HiGHS_CMD(path=highs_path, timeLimit=60, msg=False)
solver_cbc = pl.COIN_CMD(gapRel = 0.01, timeLimit =60, msg=True)
```
and I have been getting this error :

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
/var/folders/6k/9f7wbmjs2c10jwn2q3f30mt40000gn/T/ipykernel_9417/1301324021.py in <module>
----> 1 from iitm_scheduler import wntrModel, stateGenerate, generateSchedule, generateFeasible, generateBestLP, plotSchedule, plotTankLevel

~/Downloads/voila-authentication/WDN_Scheduler/iitm_scheduler.py in <module>
     16
     17 highs_path = "/Users/surbhisharma/Downloads/voila-authentication/WDN_Scheduler/HiGHS/build/bin/highs"
---> 18 solver_highs = pl.HiGHS_CMD(path=highs_path, timeLimit=60, msg=False)
     19 solver_cbc = pl.COIN_CMD(gapRel = 0.01, timeLimit =60, msg=True)
     20

AttributeError: module 'pulp' has no attribute 'HiGHS_CMD'
AttributeError: module 'pulp' has no attribute 'HiGHS_CMD'
```
Things I have done are:
I have download the HiGHS using source code using [HiGHS Documentation](https://ergo-code.github.io/HiGHS/get-started.html#Download-source-code) and then also cloned [PuLP](https://github.com/coin-or/pulp/). I can run the `highs` command after installation but still when I am trying to run the Jupyter notebook it gives the same error. 

Can someone please let me know why doesn't this work or what is it that might be doing wrong?
Thanks!

**Comment by jajhall** [comment]
I don't think that this is a HiGHS issue, and I don't know whether it's supposed to work.

However (in https://github.com/coin-or/pulp/issues/553) I've indicated that we're happy to coordinate with work on improving `highs_api.py`

**Comment by berkorbay** [comment]
It is simply because PuLP's latest release is on Dec 21, 2021 and they added HiGHS interface in February. Simply install from Github.

`python -m pip install -U git+https://github.com/coin-or/pulp`

**Comment by jajhall** [comment]
> python -m pip install -U git+https://github.com/coin-or/pulp

Working in Linux (up-to-date Ubuntu) I've done this, and successfully solved a MIP with Cbc and, by switching solvers, Gurobi 

However, my `HiGHS.py`

> path_to_highs = r'\home\jajhall\HiGHS\build\bin\highs'
> import pulp as pl
> print(pl.listSolvers(onlyAvailable=True))
> solver = pl.HIGHS_CMD(path=path_to_highs)

yields

> >> python3 HiGHS.py 
> Traceback (most recent call last):
> ['GUROBI_CMD', 'PULP_CBC_CMD']
>   File "HiGHS.py", line 5, in <module>
>     solver = pl.HIGHS_CMD(path=path_to_highs)
> AttributeError: module 'pulp' has no attribute 'HIGHS_CMD'

Anything obvious that I need to do?

Since I'd like to update `highs_api.py`, maybe I need to fork PuLP from Github anyway

**Comment by berkorbay** [comment]
I just used the following line and it worked. HiGHS support there is kind of new and the API is through the command line I think.

`solver = pl.getSolver('HiGHS_CMD',keepFiles=False,msg=False,timeLimit=30)`

**Comment by jajhall** [comment]
Thanks: I see that it's 'HiGHS_CMD', not 'HIGHS_CMD'!

**Comment by jajhall** [comment]
I've got PuLP to use HiGHS to solve a simple LP. There's an error in parsing the solution file, but I see the `highs_api.py` file that's being used and that my edits to it are reflected in output. Hurrah for non-compiled programming languages! :D

**Comment by jajhall** [comment]
I was puzzled by this
> warnings.warn("HiGHS_CMD cannot solve the relaxation of a problem")
but guess that this is executed if a user has indicated in PuLP that they want to solve the LP relaxation of a MIP. HiGHS can solve the relaxation by specifying
                cmd += " --solver simplex"

Also, with respect to
>        if lp.sense == constants.LpMaximize:
>               # we swap the objectives
>               # because it does not handle maximization.
>               warnings.warn(
>                   "HiGHS_CMD does not currently allow maximization, "
 >                  "we will minimize the inverse of the objective function."
>               )
>               lp += -lp.objective
HiGHS can handle maximization. All PuLP has to do is communicate this is have the
> OBJSENSE
>  MAX
section in the MPS file

**Comment by ssurbhi560** [comment]
Thanks @jajhall and @berkorbay for looking into this! :)

Just leaving a note here on how I fixed this for my problem :

I couldn't use this command `python -m pip install -U git+https://github.com/coin-or/pulp` for some reason in the VM I was working in. It got stuck and just kept showing this :

```

Collecting git+https://github.com/coin-or/pulp
  Cloning https://github.com/coin-or/pulp to /tmp/pip-req-build-qclcqpy6
  Running command git clone -q https://github.com/coin-or/pulp /tmp/pip-req-build-qclcqpy6
```

It is probably related to pip -- see [this](https://github.com/pypa/pip/issues/8876)

So, what I ended up doing was, I cloned the PuLP repository using `git clone git@github.com:coin-or/pulp.git` and then after doing `cd PuLP` I ran `pip install .` After doing this I could run the jupyter notebook I was trying to run.
